store/views.py

```diff
@@ -77,6 +165,56 @@ def loanBookView(request):
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    response_data = {
+        'message': None,
+    }
+
+    if request.method == "POST":
+        BookId = request.POST.get('bookid')
+        ReturnedBookCopy = BookCopy.objects.get(id = BookId)
+        response_data['message'] = "success"
+        ReturnedBookCopy.status = True
+        ReturnedBookCopy.borrow_date = None
+        ReturnedBookCopy.borrower = None
+        ReturnedBookCopy.save()
+    
+    return JsonResponse(response_data)
+
+
+@csrf_exempt
+@login_required
+def rateBookView(request):
+    response_data = {
+        'message': None,
+    }
+    if request.method == "POST":
+        response_data['message'] = "success"
+        bookId = request.POST.get('bookid')
+        rating = request.POST.get('ratingOfTheBook')
+        BookWhichIsRated = Book.objects.get(id = bookId) 
+        PreviousRatingOfThisUser = RatingOfBook.objects.filter(book = BookWhichIsRated, userWhoRated = request.user)
+        if PreviousRatingOfThisUser:
+            
+            for x in PreviousRatingOfThisUser:
+                x.ratingValue = rating
+                x.save()
+
+        else:    
+            RatingToBeAdded = RatingOfBook()
+            RatingToBeAdded.book = BookWhichIsRated
+            RatingToBeAdded.ratingValue = rating
+            RatingToBeAdded.userWhoRated = request.user
+            RatingToBeAdded.save()
+        # update rating of the book in the Book model
+        Ratings = [x for x in RatingOfBook.objects.filter(book = BookWhichIsRated)]
+        l = len(Ratings)
+        sum = 0
+        for r in Ratings:
+            sum = sum + r.ratingValue
 
+        average_rating = sum/l
+        BookWhichIsRated.rating = average_rating
+        BookWhichIsRated.save()
+            
```

Comment:
While this is correct , a more clean and Django-ish way for doing this could be:
It uses aggeregate function in Django. You can read more about it in their documentation.
```python
average_rating = RatingOfBook.objects.filter(book__pk=book_id).aggregate(Avg('rating'))['rating__avg']
```